### DISCUSSION TITLE
usage beans for component parameters

### DISCUSSION THREAD

**Discussion by denibrek** [apache/camel-karavan#785] [Q&A]
in camel-karavan beans are quite often used to provide complex parameters for the components. This works pretty well when parameter needs to be assigned with a class instance (e.g. ConnectionFactory, etc.).

But some components may require more special parameters. E.g. for AS2 component there is a 'encryptingCertificateChain' which is java.security.cert.Certificate[].
The Karavan designer allows selecting bean for this parameter, but how would it be possible to configure the bean for this?

**Comment by davsclaus** [comment]
IMHO its better to make the camel components easier to configure, so if it requries an array of X then what we usually do is to make this able to configure as a String where you can then refer to X+ more easily. 

However camel-as2 is donated by end user whom may not have that in mind, so some components are not as easily configureable.

**Reply by denibrek** [reply]
It looks like setting array properties is in general significant problem with YAML DSL.
Shortly I have tried to create a bean like following:
`  - name: constraint`
`    properties:`
`      name: BASIC`
`      authenticate: 'true'`
`      roles: [Users, Administrators]`
`    type: org.eclipse.jetty.util.security.Constraint`

roles is a string array property.
I guess I have tried all Yaml array notations - no success.

**Comment by mgubaidullin** [comment]
@davsclaus how this ^^^ parameters could be set in YAML DSL?
or is it better to make a custom Kamelet on top of the component?

**Reply by davsclaus** [reply]
Yeah camel-as2 is an example of how NOT to do a component. Yeah either we improve the component, or make a kamelet to hide some of this pain

**Comment by davsclaus** [comment]
https://issues.apache.org/jira/browse/CAMEL-21863
